app/views/projects/versions/view.scala.html

```diff
@@ -135,6 +136,14 @@ <h1 class="pull-left">@v.v.versionString</h1>
                                     </button>
                                     <ul class="dropdown-menu" aria-labelledby="admin-version-actions">
                                         <li><a href="@appRoutes.showLog(None, None, None, v.v.id, None, None, None)">User Action Logs</a></li>
+                                        @if(request.data.globalPerm(ReviewProjects)) {
+                                            @if(v.v.visibility == VisibilityTypes.SoftDelete) {
+                                                <li><a href="#" data-toggle="modal" data-target="#modal-undelete">Undo delete</a></li>
+                                            }
+                                            @if(request.data.globalPerm(HardRemoveVersion) && !v.isRecommended) {
```

Comment:
Only folks with hard delete perm should be Ore admins, and they should be allowed to delete even if it's a recommended version.

Reply:
Its a bit more complicated to delete a recommended version the reason I did it like this is because a project must *always* have 1 public version (thats just the way Ore works) so to hard delete a version you should mark another build as recommended or soft delete it first (soft delete will set the second to last released version as recommended) and then hard delete. This forbids the bad practice of deleting the last public version (and breaking Ore) instead of removing the project.

Reply:
But by the time you get to hard delete, you've already soft deleted, and thus the version should have been bumped from recommended?

Reply:
Yes, so this check is only a sanity check.